### PR TITLE
[eigen3] Fix eigen3 and cuda compatibility issue

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix issue when using eigen3:
```
f:\vcpkg\installed\x64-windows\include\eigen3\eigen\src/Core/arch/CUDA/Half.h(212): error: more than one instance of overloaded function "__hadd" matches the argument list:
            function "__hadd(int, int)"
            function "__hadd(__half, __half)"
            argument types are: (const Eigen::half, const Eigen::half)
```

Using the official patch: https://eigen.tuxfamily.org/bz/show_bug.cgi?id=1526

### CODE REVIEW

File: ports/eigen3/portfile.cmake [hunk 1]
```diff
@@ -29,16 +28,13 @@ vcpkg_install_cmake()
...
 vcpkg_fixup_pkgconfig()
 
 file(GLOB INCLUDES ${CURRENT_PACKAGES_DIR}/include/eigen3/*)
-# Copy the eigen header files to conventional location for user-wide MSBuild integration
 file(COPY ${INCLUDES} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
 
-# Put the licence file where vcpkg expects it
-file(COPY ${SOURCE_PATH}/COPYING.README DESTINATION ${CURRENT_PACKAGES_DIR}/share/eigen3)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/eigen3/COPYING.README ${CURRENT_PACKAGES_DIR}/share/eigen3/copyright)
+file(INSTALL ${SOURCE_PATH}/COPYING.README DESTINATION ${CURRENT_PACKAGES_DIR}/share/eigen3/copyright)
```
Comment (JackBoosY) [inline ports/eigen3/portfile.cmake:40]:
```suggestion
file(INSTALL ${SOURCE_PATH}/COPYING.README DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Review by NancyLi1013 — APPROVED** [review]

File: ports/eigen3/portfile.cmake [hunk 2]
```diff
@@ -9,6 +7,7 @@ vcpkg_from_gitlab(
     REF 3.3.7
     SHA512 4cc3717b9cbe78335e05f724919497214edd482d4812aeb1a9fd6da5b3f6d1b194bb93ed0dab9e734b4334a5b88e8f8c339c43c1b2044332286ef5e758f9ecf4
     HEAD_REF master
+    PATCHES fix-cuda-error.patch
```
Comment (ras0219-msft) [inline ports/eigen3/portfile.cmake:10]: Is this issue posted upstream? If so, could you add a comment here linking that issue?

Comment (JackBoosY) [inline ports/eigen3/portfile.cmake:10]:
```suggestion
    PATCHES fix-cuda-error.patch # issue https://gitlab.com/libeigen/eigen/-/issues/1526
```

Comment (JackBoosY) [reply ports/eigen3/portfile.cmake]: @ras0219-msft Done.

**Review by strega-nil — APPROVED** [review]

**Pull request merged**